test/Driver/Dependencies/moduleonly.swift

```diff
@@ -0,0 +1,65 @@
+// RUN: test ! -f %t/buildrecord.swiftdeps
+// RUN: test   -f %t/buildrecord.swiftdeps~moduleonly
+
+// CHECK1-DAG: -primary-file ./foo.swift
+// CHECK1-DAG: -primary-file ./bar.swift
+// CHECK1-DAG: -primary-file ./baz.swift
+
+// RUN: cd %t && %swiftc_driver -c -emit-module -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1 | %FileCheck -check-prefix=CHECK2 %s
+// RUN: test -f %t/buildrecord.swiftdeps
+// RUN: test -f %t/buildrecord.swiftdeps~moduleonly
+
+// CHECK2-DAG: -primary-file ./foo.swift
+// CHECK2-DAG: -primary-file ./bar.swift
+// CHECK2-DAG: -primary-file ./baz.swift
+
+// RUN: cd %t && %swiftc_driver -emit-module -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1 | %FileCheck -check-prefix=CHECK3 %s
+// RUN: test -f %t/buildrecord.swiftdeps~moduleonly
+// RUN: test -f %t/buildrecord.swiftdeps
+
+// CHECK3-NOT: -primary-file ./foo.swift
+// CHECK3-NOT: -primary-file ./bar.swift
+// CHECK3-NOT: -primary-file ./baz.swift
+
+// RUN: touch -t 201801230123 %t/bar.swift
+// RUN: cd %t && %swiftc_driver -emit-module -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1 | %FileCheck -check-prefix=CHECK4 %s
+
+// CHECK4-NOT: -primary-file ./foo.swift
+// CHECK4-NOT: -primary-file ./baz.swift
+// CHECK4-DAG: -primary-file ./bar.swift
+
+// RUN: touch -t 201801230145 %t/baz.swift
+// RUN: cd %t && %swiftc_driver -c -emit-module -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1 | %FileCheck -check-prefix=CHECK5 %s
+
+// CHECK5-NOT: -primary-file ./foo.swift
+// CHECK5-DAG: -primary-file ./bar.swift
+// CHECK5-DAG: -primary-file ./baz.swift
+
+// RUN: cd %t && %swiftc_driver -emit-module -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1 | %FileCheck -check-prefix=CHECK6 %s
+
+// CHECK6-NOT: -primary-file ./foo.swift
+// CHECK6-NOT: -primary-file ./bar.swift
+// CHECK6-NOT: -primary-file ./baz.swift
+
+
+// '-c' (without '-emit-module') from clean environment.
+//
+// RUN: rm -rf %t && cp -r %S/Inputs/moduleonly/ %t
+// RUN: touch -t 201801230045 %t/*.swift
+// RUN: cd %t && %swiftc_driver -c -g -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1
+// RUN: test ! -f %t/buildrecord.swiftdeps~moduleonly
+// RUN: test   -f %t/buildrecord.swiftdeps
+// RUN: test ! -f %t/foo~partial.swiftmodule
+
+// '-emit-library -g' (without '-emit-module') from clean environment.
+//
+// RUN: rm -rf %t && cp -r %S/Inputs/moduleonly/ %t
+// RUN: touch -t 201801230045 %t/*.swift
+// RUN: cd %t && %swiftc_driver -emit-library -g -output-file-map ./output.json -incremental ./foo.swift ./bar.swift ./baz.swift -module-name testmodule -v 2>&1
```

Comment:
Oops, one last thing: these should all be `%target-build-swift` instead of `%swiftc_driver` because it's possible that the current setup can't build for host. Other than that this all looks good!